launchdarkly/client.go

```diff
@@ -0,0 +1,79 @@
+package launchdarkly
+
+import (
+	"net/http"
+	"encoding/json"
+	"bytes"
+	"io/ioutil"
+	"errors"
+	"strconv"
+)
+
+type Client struct {
+	AccessToken string
+}
+
+func (c *Client) GetStatus(path string) (int, error) {
+	expectedStatus := make(map[int]bool)
+	return c.execute("GET", path, nil, expectedStatus, nil)
+}
+
+func (c *Client) Get(path string, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("GET", path, nil, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Post(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("POST", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Patch(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("PATCH", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Delete(path string, expectedStatus map[int]bool) error {
+	_, err := c.execute("DELETE", path, nil, expectedStatus, nil)
+	return err
+}
+
+func (c *Client) execute(method string, path string, body interface{}, expectedStatus map[int]bool, response interface{}) (int, error) {
```

Comment:
Personal preference, but I find it very weird to pass a response as a parameter in a HTTP request method, especially since Golang supports multiple returns.

I would expect the return to be   `response, status, error`

Reply:
Right. I initially tried to pass the response body as a ref param because I thought the JSON stack could un-marshal JSON directly into a struct, but that didn't turn out too well. I change the code as you suggested.